### ISSUE TITLE
geoDisplay cannot display DDCMS geometries

### ISSUE THREAD

**Issue by rsehgal** [AIDASoft/DD4hep#511]
Hello Developers,
I am trying to visualize the geometries created using DD4hep  using geoDisplay.
i am able to visualize geometries of test example. But when i try to visualize geometries of DDCMS, then its give error relate the document type of DDDefinition.

Could you please suggest how can i display DDCMS geometries,

 - OS version: Ubuntu 18.04
 - Compiler version:  GCC 7.2
 - DD4hep version: tag 01-10
 - Goal: mentioned above

**Comment by petricm** [comment]
Please provide what command you are executing and the error you get

**Comment by rsehgal** [comment]
After compiling the example i am using following command
geoDisplay -compact tracker.xml
(tracker.xml is in the current directory) 

error message is as follows:
**************************************************************************************************************
PersistencyIO    INFO  +++ Set Streamer to dd4hep::OpaqueDataBlock
Info in <TGeoManager::TGeoManager>: Geometry default, Detector Geometry created
CompactLoader    INFO  +++ Processing compact file: tracker.xml with flag BUILD_DEFAULT
RunPlugin        ERROR ++ Exception while executing plugin DD4hep_CompactLoader:
		dd4hep: Failed to locate plugin to interprete files of type "DDDefinition" - no factory:DDDefinition_XML_reader. 		No factory with name Create(DDDefinition_XML_reader) for type DDDefinition_XML_reader found.
		Please check library load path and/or plugin factory name.
dd4hep: while parsing tracker.xml
dd4hep: with plugin:DD4hep_CompactLoader
geoPluginRun: Got uncaught exception: RunPlugin: ++ Exception while executing plugin DD4hep_CompactLoader:
		dd4hep: Failed to locate plugin to interprete files of type "DDDefinition" - no factory:DDDefinition_XML_reader. 		No factory with name Create(DDDefinition_XML_reader) for type DDDefinition_XML_reader found.
		Please check library load path and/or plugin factory name.
dd4hep: while parsing tracker.xml
dd4hep: with plugin:DD4hep_CompactLoader
**************************************************************************************************************

**Comment by rsehgal** [comment]
Hello Developers,
Any suggestions on the above mentioned problem??

**Comment by andresailer** [comment]
Did you compile the examples/DDCMS libraries?

**Comment by rsehgal** [comment]
Yes i did, and also checked the libDDCMS.so got created.

So once the library is generated, how can i visualized the geometry
like for other dd4hep application it is like **geoDisplay -compact test.xml**

What i read about DDCMS is that its just the converter of CMS based file structures
So once the library is ready how to use it to visualize it, or to use in the simulations

**Comment by andresailer** [comment]
Do you have the folder, where libDDCMS.so is created in the LD_LIBRARY_PATH? There should also be a `components` file in that folder, which contains the line `libDDCMS.so:DDDefinitions_XML_reader` , which is used by the plugin service to find the library to load.

**Comment by petricm** [comment]
or try sourcing the `thisDDCMS.sh` in the install directory before calling geoDispaly

**Comment by MarkusFrankATcernch** [comment]
This is the command to display the CMS tracker geometry:
geoDisplay -compact file:<path>/DDCMS/data/cms_tracker.xml

Visualization is not included here. You only get the "default colors".
And of course: the library must be in the LD_LIBRARY_PATH

**Comment by rsehgal** [comment]
Thanks a lot, with the help of above suggestions, now i am able to visualize the the cms_tracker.

**Comment by rsehgal** [comment]
Another question :
To understand clearly, can we have a very simple example of detector construction along with xml for
DDDefintion which will create a two coaxial tubes.

What i understood is for detector constructor we need need to have Algorithm part in xml file for DDCMS, then what is the use of LogicalPartSection and PosPartSection.

So if we have simple example for DDCMS then it will be very much useful to understand the framework. 

May i request developers to provide a minimal DDCMS specific example with bare minimal xml file.
Which will also allow the learners to understand the framework.

**Comment by andresailer** [comment]
Please open a new issue for a new problem

**Comment by rsehgal** [comment]
Thanks for all the replies. 
Now able to visualize cms tracker,
 hence closing the issue
